modules/ROOT/pages/external-oauth-2.0-token-validation-policy.adoc

```diff
@@ -1,43 +1,173 @@
+The following diagram illustrates the OAuth 2.0 Token Enforcement using Mule OAuth Provider policy workflow:
 
-Note the following requirements and restrictions:
+image:open-id-policy-workflow.png[OAuth 2.0 Token Enforcement Workflow]
 
-* To use the OAuth 2.0 Access Token Enforcement Using External Provider policy, you need a Mule OAuth 2.0 provider to provide an access token. You can't use any other OAuth 2.0 provider, such as Facebook, Google, or Azure.
-* Like other API Manager-enforced policies, the API needs to be registered in API Manager to apply and use any OAuth 2.0 access token policy.
-* None of the token enforcement policies work with a Mule client app to access OAuth 2.0-protected resources outside Anypoint Platform. Configure the HTTP Requester connector for this purpose.
+As shown in the diagram:
 
-== Prerequisites
+. The user first sends an HTTP request to the API protected by the policy. 
+. The policy extracts the token from the request and sends it to the validation endpoint to verify the integrity of the token.
+. The token validation endpoint returns token metadata, including the client ID of the client application.
+. Using a local database updated with contracts that were previously obtained from Anypoint Platform, the policy verifies whether the client ID has access to the API. 
+. If all the validations are successfully completed, the request is allowed to reach the backend.
 
-You are an Anypoint Platform organization administrator or have permission to create or manage APIs in an environment.
+[[leveraging-TVE-information]]
+=== Leveraging Token Validation Endpoint Information for Mule Applications
+
+When a token validation endpoint successfully validates a token, certain information, such as configurable fields from Mule OAuth provider, are returned to the policy.
+
+The following example illustrates the token validation response from the authorization server:
+
+[source,text,linenums]
+---
+{
+    "uid":"john.doe",
+    "mail":"john.doe@example.com"
+    "token_type":"Bearer",
+    "expires_in":3600,
+    "alias" : ["Jhon", "Jhonny", "Mr Doe"],
+    "address" : {
+        "city": "london", 
+        "road": “abbey road"
+    }
+    
+}
+---
+The fields returned by the OAuth provider are processed by the policy, propagated throughout the Mule flow, and finally exposed to the backend if the application requesting the access uses an HTTP requester.
+
+=== Token Caching
+
+After the system validates a token, that token is cached (by default) until it expires, to improve the performance of the policy. In certain cases, such as in the case of revoked tokens, you might want to minimize or even disable caching.
+
+To control the time taken to cache the tokens before retrying a validation against the token validation endpoint, specify the following property when starting the Mule runtime engine:
```

Comment:
Is this more clear?
```suggestion
To control the time that invalid tokens are cached before attempting revalidation, specify the following property when starting the Mule runtime engine:
```
I did not at first understand why one would retry validation if the token is validated, as in the previous sentence.

Reply:
To control the time taken to cache the tokens, before you attempt revalidating against the token validation endpoint, specify the following property when starting the Mule runtime engine:

Done! Thanks!